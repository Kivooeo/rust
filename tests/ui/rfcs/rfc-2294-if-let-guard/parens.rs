// Parenthesised let "expressions" are not allowed in guards

//@revisions: edition2021 edition2024
//@[edition2021] edition:2021
//@[edition2024] edition:2024

#![feature(if_let_guard)]

#[cfg(false)]
fn un_cfged() {
    match () {
        () if let 0 = 1 => {}
        () if (let 0 = 1) => {}
        //~^ ERROR expected expression, found `let` statement
        () if (((let 0 = 1))) => {}
        //~^ ERROR expected expression, found `let` statement
    }
}

fn main() {
    match () {
        () if let 0 = 1 => {}
        () if (let 0 = 1) => {}
        //~^ ERROR expected expression, found `let` statement
        () if (((let 0 = 1))) => {}
        //~^ ERROR expected expression, found `let` statement
    }
}
