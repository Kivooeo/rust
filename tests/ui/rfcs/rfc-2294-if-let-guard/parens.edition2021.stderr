error: expected expression, found `let` statement
  --> $DIR/parens.rs:13:16
   |
LL |         () if (let 0 = 1) => {}
   |                ^^^^^^^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions
note: `let`s wrapped in parentheses are not supported in a context with let chains
  --> $DIR/parens.rs:13:16
   |
LL |         () if (let 0 = 1) => {}
   |                ^^^^^^^^^

error: expected expression, found `let` statement
  --> $DIR/parens.rs:15:18
   |
LL |         () if (((let 0 = 1))) => {}
   |                  ^^^^^^^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions
note: `let`s wrapped in parentheses are not supported in a context with let chains
  --> $DIR/parens.rs:15:18
   |
LL |         () if (((let 0 = 1))) => {}
   |                  ^^^^^^^^^

error: expected expression, found `let` statement
  --> $DIR/parens.rs:23:16
   |
LL |         () if (let 0 = 1) => {}
   |                ^^^^^^^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions
note: `let`s wrapped in parentheses are not supported in a context with let chains
  --> $DIR/parens.rs:23:16
   |
LL |         () if (let 0 = 1) => {}
   |                ^^^^^^^^^

error: expected expression, found `let` statement
  --> $DIR/parens.rs:25:18
   |
LL |         () if (((let 0 = 1))) => {}
   |                  ^^^^^^^^^
   |
   = note: only supported directly in conditions of `if` and `while` expressions
note: `let`s wrapped in parentheses are not supported in a context with let chains
  --> $DIR/parens.rs:25:18
   |
LL |         () if (((let 0 = 1))) => {}
   |                  ^^^^^^^^^

error: aborting due to 4 previous errors

